lib/Runtime/Library/InJavascript/Intl.js

```diff
@@ -1593,44 +1593,122 @@
                 dateTimeFormat.__initializedDateTimeFormat = true;
             }
 
-            platform.registerBuiltInFunction(tagPublicFunction("Date.prototype.toLocaleString", function () {
-                if (typeof this !== 'object' || !(this instanceof Date)) {
-                    platform.raiseNeedObjectOfType("Date.prototype.toLocaleString", "Date");
-                }
-                var value = callInstanceFunc(DateInstanceGetDate, new Date(this));
-                if (isNaN(value) || !isFinite(value)) {
-                    return "Invalid Date";
-                }
-                var stateObject = setPrototype({}, null);
-                InitializeDateTimeFormat(stateObject, arguments[0], ToDateTimeOptions(arguments[1], "any", "all"));
-                return String(platform.formatDateTime(Internal.ToNumber(this), stateObject));
-            }), 0);
+            {
+                // scope for the cache variable
+                let __date_prototype_toLocaleString_default_formatterStateObject_cache = undefined;
+                platform.registerBuiltInFunction(tagPublicFunction("Date.prototype.toLocaleString", function () {
```

Comment:
curious - why not define locale and options as formals?

Reply:
The length of the function is required to be 0 by spec (and the length is equal to the number of formals), so we have to get arguments this way. This is done elsewhere. I'll add comments.

Reply:
Spoke offline -- We can use default params to capture these values in a more canonical way and keep the function length as intended. I'll make this update.